pkg/status/conditions.go

```diff
@@ -0,0 +1,200 @@
+//
+// +kubebuilder:validation:Type=array
+type Conditions map[ConditionType]Condition
+
+// NewConditions initializes a set of conditions with the given list of
+// conditions.
+func NewConditions(conds ...Condition) Conditions {
+	conditions := Conditions{}
+	for _, c := range conds {
+		conditions.SetCondition(c)
+	}
+	return conditions
+}
+
+// IsTrue searches the set of conditions for a condition with the given
+// ConditionType. If found, it returns `condition.IsTrue()`. If not found,
+// it returns false.
+func (conditions Conditions) IsTrue(t ConditionType) bool {
+	if condition, ok := conditions[t]; ok {
+		return condition.IsTrue()
+	}
+	return false
+}
+
+// IsFalse searches the set of conditions for a condition with the given
+// ConditionType. If found, it returns `condition.IsFalse()`. If not found,
+// it returns false.
+func (conditions Conditions) IsFalse(t ConditionType) bool {
+	if condition, ok := conditions[t]; ok {
+		return condition.IsFalse()
+	}
+	return false
+}
+
+// IsUnknown searches the set of conditions for a condition with the given
+// ConditionType. If found, it returns `condition.IsUnknown()`. If not found,
+// it returns true.
+func (conditions Conditions) IsUnknown(t ConditionType) bool {
+	if condition, ok := conditions[t]; ok {
+		return condition.IsUnknown()
+	}
+	return true
+}
+
+// SetCondition adds (or updates) the set of conditions with the given
+// condition. It returns a boolean value indicating whether the set condition
+// is new or was a change to the existing condition with the same type.
+func (conditions *Conditions) SetCondition(newCond Condition) bool {
+	if conditions == nil || *conditions == nil {
+		*conditions = make(map[ConditionType]Condition)
+	}
+	newCond.LastTransitionTime = metav1.Time{Time: clock.Now()}
+
+	if condition, ok := (*conditions)[newCond.Type]; ok {
+		if condition.Status == newCond.Status {
+			newCond.LastTransitionTime = condition.LastTransitionTime
+		}
+		changed := condition.Status != newCond.Status ||
+			condition.Reason != newCond.Reason ||
+			condition.Message != newCond.Message
+		(*conditions)[newCond.Type] = newCond
+		return changed
+	}
+	(*conditions)[newCond.Type] = newCond
+	return true
+}
+
+// GetCondition searches the set of conditions for the condition with the given
+// ConditionType and returns it. If the matching condition is not found,
+// GetCondition returns nil.
+func (conditions Conditions) GetCondition(t ConditionType) *Condition {
+	if condition, ok := conditions[t]; ok {
+		return &condition
+	}
+	return nil
+}
+
+// RemoveCondition removes the condition with the given ConditionType from
+// the conditions set. If no condition with that type is found, RemoveCondition
+// returns without performing any action.
+func (conditions *Conditions) RemoveCondition(t ConditionType) bool {
+	if conditions == nil || *conditions == nil {
+		return false
+	}
+	if _, ok := (*conditions)[t]; ok {
+		delete(*conditions, t)
```

Comment:
`delete` is a no-op if `t` isn’t in `conditions`, so you can just call this line without a conditional.

Reply:
Yeah, the return value is based on the presence, hence the check. I'll update the GoDoc comment though, so thanks for the reminder :)